torch/distributed/__init__.py

```diff
@@ -19,4 +20,28 @@ def is_available():
     # this.
     from .distributed_c10d import _backend  # noqa: F401
     if sys.version_info >= (3, 0):
+        from .rpc import _init_rpc
         from .rpc import *  # noqa: F401
+
+        def init_model_parallel(worker_name, worker_id, rpc_backend=RpcBackend.PROCESS_GROUP):
+            r"""
+                Initializes model parallel primitives such as the local rpc agent
+                and distributed autograd.
+
+                Initializes the local RPC agent which immediately makes the current process
+                ready to send and receive RPCs. The caller needs to make sure the specified
+                backend is properly intialized before calling this method. For example, to
+                use ``pg`` (ProcessGroup) backend, ``init_process_group`` must be invoked
+                prior to this method.
+
+                Arguments:
+                    worker_name (str): a globally unique name of this node. (e.g.,
+                                       ``Trainer3``, ``ParameterServer2``, ``Master``, ``Worker1``)
+                    worker_id (int): a globally unique id for this node. Typically, can use the
```

Comment:
does this have to be specified by users, can the backend instances figure this out on their own? Users wouldn't need this id any more this point, so it seems like this should be a purely internal variable?

Reply:
This is possible for the ProcessGroup backend, but I'm not sure how the thrift backend would figure this out automatically. Also, note that even for the ProcessGroup backend the user does have to specify this when they initialize the ProcessGroup.